src/Components/WebAssembly/WebAssembly.Authentication/src/Models/InteractiveRequestOptions.cs

```diff
@@ -0,0 +1,138 @@
+            ReturnUrl = returnUrl,
+            Scopes = null
+        };
+    }
+
+    /// <summary>
+    /// Creates a new <see cref="InteractiveRequestOptions"/> for signing in with the given return url and scopes.
+    /// </summary>
+    /// <param name="returnUrl">The URL to return to after the interactive operation.</param>
+    /// <param name="scopes">The scopes to request interactively.</param>
+    /// <returns>An <see cref="InteractiveRequestOptions"/> configured for requesting a token interactively.</returns>
+    public static InteractiveRequestOptions GetToken(string returnUrl, IEnumerable<string> scopes)
+    {
+        return new InteractiveRequestOptions
+        {
+            Interaction = InteractionType.GetToken,
+            ReturnUrl = returnUrl,
+            Scopes = scopes
+        };
+    }
+
+    /// <summary>
+    /// Creates a new <see cref="InteractiveRequestOptions"/> for signing in with the given return url and the default scopes.
+    /// </summary>
+    /// <param name="returnUrl">The URL to return to after the interactive operation.</param>
+    /// <returns>An <see cref="InteractiveRequestOptions"/> configured for requesting a token interactively.</returns>
+    public static InteractiveRequestOptions GetToken(string returnUrl)
+    {
+        return new InteractiveRequestOptions
+        {
+            Interaction = InteractionType.GetToken,
+            ReturnUrl = returnUrl,
+            Scopes = null
+        };
+    }
+
+    /// <summary>
+    /// Gets the redirect URL this request must return to upon successful completion.
+    /// </summary>
+    [JsonInclude]
+    public string ReturnUrl { get; init; }
+
+    /// <summary>
+    /// Gets the scopes this request must use in the operation.
+    /// </summary>
+    [JsonInclude]
+    public IEnumerable<string> Scopes { get; init; }
+
+    /// <summary>
+    /// Gets the request type.
+    /// </summary>
+    [JsonInclude]
+    public InteractionType Interaction { get; init; }
+
+    /// <summary>
+    /// Gets or sets the additional parameters to pass in to the underlying provider.
+    /// </summary>
+    /// <remarks>
+    /// The underlying provider is free to apply these parameters as it sees fit or ignore them completely. In the default
+    /// implementations the parameters will be JSON serialized using System.Text.Json and passed as a parameter to the
+    /// underlying JavaScript implementation that handles the operation details.
+    /// </remarks>
+    public IDictionary<string, object> AdditionalRequestParameters { get; set; } = new Dictionary<string, object>();
+
+    [UnconditionalSuppressMessage(
+        "Trimming",
+        "IL2026:Members annotated with 'RequiresUnreferencedCodeAttribute' require dynamic access otherwise can break functionality when trimming application code",
+        Justification = "Serializes 'InteractiveAuthenticationRequest' into a string")]
+    internal string ToState() => JsonSerializer.Serialize(this);
```

Comment:
Is it definitely safe to suppress the trim warning here and below? Is it possible to use the overload of `JsonSerializer.Serialize` that takes the context parameter so it uses the sourcegenerated version of the serializer/deserializer and hence always does the right thing with trimming?

Reply:
@SteveSandersonMS this is something I wasn't sure about.

We might need APIs to add to the dictionary instead, so that the trimmer can know that it needs to keep those parameters, so instead of having an `AdditionalRequestParameters` dictionary, we would have 3 APIs:
* bool TryAddAdditionalParameter<TParameter>("name", value);
* bool RemoveAdditionalParameter<TParameter("name");
* bool TryGetAdditionalParameter<TParameter>("name");

That's where I started, but I felt it was a bit cumbersome compared to a dictionary, however given the serialization constraints I think it might be best.

Reply:
I see you've done those changes now. It looks totally reasonable to me.